lib/trino-filesystem-s3/src/main/java/io/trino/filesystem/s3/S3FileSystemConfig.java

```diff
@@ -536,4 +537,17 @@ public S3FileSystemConfig setSupportsExclusiveCreate(boolean supportsExclusiveCr
         this.supportsExclusiveCreate = supportsExclusiveCreate;
         return this;
     }
+
+    public String getApplicationId()
```

Comment:
I noticed that `software.amazon.awssdk.core.internal.useragent.SdkUserAgentBuilder` warns if the length is over 50. Should we enforce this?
```java
@Size(max = 50)